star-sys/STAR/source/MmapArray.cpp

```diff
@@ -64,4 +84,13 @@ MmapArray::~MmapArray() {
         munmap(prefix_mmap, PAGE);
         prefix_mmap=NULL;
     }
+
+    if (last_page_mmap) {
+        munmap(last_page_mmap, PAGE);
+        last_page_mmap=NULL;
+    }
```

Comment:
What's going on here?  Why not just mmap the entire thing all at once?  The kernel doesn't care that you made several calls to mmap.